torch/export/dynamic_shapes.py

```diff
@@ -623,22 +623,92 @@ def _compare(tree, dynamic_shapes, path):
         raise
 
 
+def _signature(f):
+    if isinstance(f, ExportedProgram):
```

Comment:
When is it possible to have ExportedProgram as input to export?

Reply:
I think if you re-export it? Not sure, but this check exists before this PR.